tests/integration/api/v1/suggest/test_suggest_top_picks.py

```diff
@@ -5,46 +5,180 @@
+            {
+                "block_id": 0,
+                "title": "Abc",
+                "url": "https://abc.test",
+                "provider": "top_picks",
+                "is_top_pick": True,
+                "is_sponsored": False,
+                "icon": "",
+                "score": 0.25,
+            },
+        ],
+        index_char_range=(4, 7),
+        query_char_limit=4,
+        firefox_char_limit=2,
+    )
+
+
+@pytest.fixture(name="backend_mock")
+def fixture_backend_mock(mocker: MockerFixture, top_picks_data: TopPicksData) -> Any:
+    """Create a Top Pick backend mock object for test."""
+    backend_mock: Any = mocker.AsyncMock(spec=TopPicksBackend)
+    backend_mock.fetch.return_value = top_picks_data
+    return backend_mock
+
+
+@pytest.fixture(name="top_picks_parameters")
+def fixture_top_picks_parameters() -> dict[str, Any]:
+    """Define Top Pick provider parameters for test."""
+    return {"name": "top_picks", "enabled_by_default": True}
+
 
 @pytest.fixture(name="providers")
-def fixture_providers() -> Providers:
+def fixture_providers(
+    backend_mock: Any, top_picks_parameters: dict[str, Any]
+) -> Providers:
     """Define providers for this module which are injected automatically."""
-    return {"top_picks": TopPicksProvider(name="top_picks", enabled_by_default=True)}
+    return {"top_picks": Provider(backend=backend_mock, **top_picks_parameters)}
 
 
-@pytest.mark.parametrize(
-    ["query", "title", "url"],
-    [
-        ("exam", "Example", "https://example.com"),
-        ("exxa", "Example", "https://example.com"),
-        ("example", "Example", "https://example.com"),
-        ("firef", "Firefox", "https://firefox.com"),
-        ("firefoxes", "Firefox", "https://firefox.com"),
-        ("mozilla", "Mozilla", "https://mozilla.org/en-US/"),
-        ("mozz", "Mozilla", "https://mozilla.org/en-US/"),
-    ],
-)
-def test_top_picks_query(client: TestClient, query: str, title: str, url: str) -> None:
+@pytest.mark.parametrize("query", ["exam", "exxa", "example"])
+def test_top_picks_query(client: TestClient, query: str) -> None:
     """Test if Top Picks provider returns result for indexed Top Pick"""
+    expected_suggestion: list[Suggestion] = [
+        Suggestion(
+            block_id=0,
+            title="Example",
+            url="https://example.com",
+            provider="top_picks",
+            is_top_pick=True,
+            is_sponsored=False,
+            score=0.25,
+            icon="",
+        )
+    ]
+
     response = client.get(f"/api/v1/suggest?q={query}")
     assert response.status_code == 200
-
     result = response.json()
-    assert len(result["suggestions"]) == 1
-    assert result
-    assert result["suggestions"][0]["url"] == url
-    assert result["suggestions"][0]["title"] == title
-    assert result["suggestions"][0]["is_top_pick"]
-    assert not result["suggestions"][0]["is_sponsored"]
+    assert Suggestion(**result["suggestions"][0]) == expected_suggestion[0]
```

Comment:
```suggestion
    assert [Suggestion(**suggestion) for suggestion in result["suggestions"]] == expected_suggestion
```

Reply:
Good call Nan, to take it further and avoid the assertion loop, you could do:
```suggestion
    actual_suggestions: list[Suggestion] = [
        Suggestion(**suggestion) for suggestion in result["suggestions"]
    ]
    assert actual_suggestions == expected_suggestion
```

Reply:
👍 makes sense, the 0th index is the only piece of data anyway but this is indeed easier to read.